### PR TITLE
Fix input not working in Ctrl-G mode after UI Editor is opened

### PR DESCRIPTION
**Pull request by michabr**
The underlying issue was that the mouse and keyboard were being re-registered via RegisterRawInputDevices from the LyShine module. This occurred because the static variable that tracks whether a device has been registered wasn't persisting across dlls.

The RenderViewportWidget, which is used in the Main Editor viewport as well as the UI Editor viewport, creates new input devices when not in game mode in order to handle correct input event ordering between AzFramework input events and Qt events.

Signed-off-by: abrmich <abrmich@amazon.com>

### CODE REVIEW

**Review by lawsonamzn — APPROVED** [review]
I take it that this doesn't leak the new az_envirronment varible itself.

Also, isn't there already some singleton manager this int could live on instead of a new env var?

Still approving - looks ok

**Review by NicholasVanSickle — APPROVED** [review]
Good spot on the DLL boundary issue - looks good to me!

**Review by cgalvan — APPROVED** [review]

**Comment by michabr** [comment]
@lawsonamzn maybe? I didn't see any obvious singleton manager this could be added to (it's a platform dependent implementation), but maybe @bosnichd might have a better idea.

**Comment by lawsonamzn** [comment]
> @lawsonamzn maybe? I didn't see any obvious singleton manager this could be added to (it's a platform dependent implementation), but maybe @bosnichd might have a better idea.


feels like its very windows specific anyway, so maybe this is the right place for it. only concern is if it leaks the Environment object iself or whetheri t matters that it does. normally you would call .reset() on the env var on shutdown.

File: Code/Framework/AzFramework/Platform/Windows/AzFramework/Input/Devices/Mouse/InputDeviceMouse_Windows.cpp [hunk 1]
```diff
@@ -7,6 +7,7 @@
  */
 
 #include <AzCore/PlatformIncl.h>
+#include <AzCore/Module/Environment.h>
```
Comment (bosnichd) [inline Code/Framework/AzFramework/Platform/Windows/AzFramework/Input/Devices/Mouse/InputDeviceMouse_Windows.cpp:10]: Does InputDeviceKeyboard_Windows.cpp also need this #include?

File: Code/Framework/AzFramework/Platform/Windows/AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard_Windows.cpp [hunk 2]
```diff
@@ -149,6 +157,7 @@ namespace AzFramework
             AZ_Assert(result, "Failed to deregister raw input device: keyboard");
             AZ_UNUSED(result);
         }
+        s_instanceCount.Set(instanceCount);
```
Comment (bosnichd) [inline Code/Framework/AzFramework/Platform/Windows/AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard_Windows.cpp:160]: If we do need to reset the environment variable, we should probably do that at the end of the 'if' block above, then wrap this in an 'else' statement.

File: Code/Framework/AzFramework/Platform/Windows/AzFramework/Input/Devices/Mouse/InputDeviceMouse_Windows.cpp [hunk 3]
```diff
@@ -137,18 +138,25 @@ namespace AzFramework
...
-            rawInputDevice.dwFlags     = 0;
-            rawInputDevice.hwndTarget  = 0;
+            rawInputDevice.usUsage = RAW_INPUT_MOUSE_USAGE;
+            rawInputDevice.dwFlags = 0;
+            rawInputDevice.hwndTarget = 0;
             const BOOL result = RegisterRawInputDevices(&rawInputDevice, 1, sizeof(rawInputDevice));
             AZ_Assert(result, "Failed to register raw input device: mouse");
             AZ_UNUSED(result);
         }
+        s_instanceCount.Set(instanceCount);
```
Comment (bosnichd) [inline Code/Framework/AzFramework/Platform/Windows/AzFramework/Input/Devices/Mouse/InputDeviceMouse_Windows.cpp:159]: As above, if we do need to reset the environment variable, we should probably do that at the end of the 'if' block above, then wrap this in an 'else' statement.

**Review by bosnichd — APPROVED** [review]
Nice find, thanks for fixing!

Note that we use the same paradigm in InputDeviceKeyboardMac if that happens to exhibit a similar problem.

File: Code/Framework/AzFramework/Platform/Windows/AzFramework/Input/Devices/Mouse/InputDeviceMouse_Windows.cpp (hunk 1 above)
Comment (michabr) [reply Code/Framework/AzFramework/Platform/Windows/AzFramework/Input/Devices/Mouse/InputDeviceMouse_Windows.cpp:10]: Yes, added.

**Review by lawsonamzn — APPROVED** [review]

**Comment by michabr** [comment]
> > @lawsonamzn maybe? I didn't see any obvious singleton manager this could be added to (it's a platform dependent implementation), but maybe @bosnichd might have a better idea.
> 
> feels like its very windows specific anyway, so maybe this is the right place for it. only concern is if it leaks the Environment object iself or whetheri t matters that it does. normally you would call .reset() on the env var on shutdown.

Added a Reset for the new environment variable so it doesn't leak (in case the device is added/removed/added/removed multiple times). Otherwise, it would stick around for the lifetime of the application. When looking at this, I noticed the system entity doesn't get freed up on app shutdown, but that should be OK.

**Comment by AMZN-alexpete** [comment]
Does this fix https://github.com/o3de/o3de/issues/2515 ?

**Comment by michabr** [comment]
> Does this fix #2515 ?

Yes.

**Pull request merged**